core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamUnitTest.java

```diff
@@ -0,0 +1,98 @@
+package com.baeldung.inputstream;
+
+import static org.junit.jupiter.api.io.CleanupMode.ALWAYS;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Assert;
```

Comment:
You are using a JUnit4 assertion instead of JUnit5. Since we have complex assertions in this test, let's use AssertJ everywhere instead + include the static imports, don't type Assertions.assertThat everywhere, instead add static assertions so you can directly say assertThat.

Reply:
Updated